### PR TITLE
Generalize pois queries for diverse tags

### PR DESCRIPTION
**Pull request by chesterharvey**
I modified the `parse_poi_query`, `osm_poi_download`, and `pois_from_place` functions so they accept a `tags` dictionary that allows querying a variety of tags, not just the 'amenity' tag, as previously implemented.

Desired tags are passed to `pois_from_place` with a dictionary like this:
```
tags = {
    'amenity':True,
    'leisure':False,
    'landuse':['retail','commercial'],
    'highway':'bus_stop'
}
gdf = ox.pois_from_place(place='Santa Clara, Santa Clara County, California, USA', tags=tags)
```
A particular key:value pair (e.g., 'highway':'bus_stop') returns elements with that particular pair. A list of values for a certain key (e.g., 'landuse':['retail','commercial']) returns all elements with any of those values. A `True` value (e.g., 'amenity':True) retrieves all elements with that key.

In theory, a `False` value (e.g., 'leisure':False) would filter out any element with that key, but this doesn't seem to work reliably. I don't know if this is an issue with my query construction or a bug with Overpass. To help troubleshoot this, I included an optional boolean parameter (`return_query`) to return the constructed Overpass query alongside the GDF. For example, this is query constructed based on the tag dictionary above:

```
'[out:json][timeout:180];((nwr[~"^(amenity)$"~".*"](37.812276,-122.249374,37.833026,-122.210148);(._;>;););(nwr["leisure"!~".*"](37.812276,-122.249374,37.833026,-122.210148);(._;>;););(nwr[landuse~"retail|commercial"](37.812276,-122.249374,37.833026,-122.210148);(._;>;););(nwr[highway~"bus_stop"](37.812276,-122.249374,37.833026,-122.210148);(._;>;);););out;'
```

### CODE REVIEW

**Comment by gboeing** [comment]
Thanks @chesterharvey.
@HTenkanen would you mind taking a glance at this PR?

File: osmnx/pois.py [hunk 1]
```diff
@@ -372,21 +443,33 @@ def pois_from_point(point, distance=None, amenities=None):
...
-        List of amenities that will be used for finding the POIs from the selected area. 
-        See available amenities from: http://wiki.openstreetmap.org/wiki/Key:amenity
+        If provided, will override tags with the dictionary {'amenities':amenities}.
+        Accommodates the depricatd amenities parameter.
 
     Returns
     -------
     GeoDataFrame 
     """
+    if 'amenities' in locals():
```
Comment (gboeing) [inline osmnx/pois.py]: Is this simpler than just leaving default argument `amenities=None` and checking `if amenities is not None` here? (Same question goes for the other pois_from_xxxx() functions.)

File: osmnx/pois.py [hunk 2]
```diff
@@ -397,59 +480,98 @@ def pois_from_address(address, distance, amenities=None):
...
+        Values make be string, lists of strings, or None, if all values should be returned for a given key.
+        By default, all POIs with an 'amenity' key of any value will be be returned.
+
+    Keyword Arguments
+    -----------------
     amenities : list
-        List of amenities that will be used for finding the POIs from the selected area. See available 
-        amenities from: http://wiki.openstreetmap.org/wiki/Key:amenity
+        If provided, will override tags with the dictionary {'amenities':amenities}.
+        Accommodates the depricatd amenities parameter.
```
Comment (gboeing) [inline osmnx/pois.py]:
```suggestion
        Accommodates the deprecated amenities parameter.
```

**Comment by gboeing** [comment]
@chesterharvey let me know if you are able to respond to the code review

**Comment by gboeing** [comment]
@chesterharvey I'm going to close this PR for now as it has gone stale and I don't have permissions to push to your branch. Happy to re-open if you have time for the code review + resolving the merge conflicts.

**Comment by kylebarron** [comment]
I was just looking at the code and was thinking about submitting a similar PR to get POIs that aren't given a tag of `amenity`.

I may submit a PR for this soon, but I'll probably try to write my own code instead of starting from this PR.

**Comment by gboeing** [comment]
superseded by #449

**Pull request closed without merging**